src/libs/ModifiedExpenseMessage.ts

```diff
@@ -215,6 +231,31 @@ function getForReportAction(reportID: string | undefined, reportAction: OnyxEntr
         });
     }
 
+    const hasModifiedTaxAmount = reportActionOriginalMessage && 'oldTaxAmount' in reportActionOriginalMessage && 'taxAmount' in reportActionOriginalMessage;
+    if (hasModifiedTaxAmount) {
+        const transactionThread = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`] ?? null;
+        const iouReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${transactionThread?.parentReportID}`] ?? null;
+        const currency = iouReport?.currency ?? '';
```

Comment:
Can we please read currency from reportActionOriginalMessage. I have updated backend to send it. Reading from IOU report is wrong because if currency is updated, system messages would have new currency and not the currency it was at the time of updating tax amount

```suggestion
        const currency = reportActionOriginalMessage?.currency;
```